src/server.rs

```diff
@@ -220,27 +233,37 @@ impl CoAPServer {
     }
 }
 
-fn transmit_handler(tx_recv: RxQueue, tx_only: UdpSocket) {
-    // Note! We should only transmit with this UDP Socket
+fn response_handler(tx_recv: &RxQueue, tx_only: &UdpSocket) {
     // TODO: Add better support for failure detection or logging
-    loop {
-        match tx_recv.recv() {
-            Ok(q_res) => {
-                match q_res.response.message.to_bytes() {
-                    Ok(bytes) => {
-                        let _ = tx_only.send_to(&bytes[..], &q_res.address);
-                    }
-                    Err(_) => {
-                        error!("Failed to decode response");
+    match tx_recv.recv() {
+        Ok(q_res) => {
+            match q_res.response {
+                Some(resp) => {
+                    match resp.message.to_bytes() {
+                        Ok(bytes) => {
+                            match tx_only.send_to(&bytes[..], &q_res.address) {
+                                Ok(None) => {
+                                    // Look at https://github.com/carllerche/mio/issues/411 in detail
+                                    error!("Failed to complete the response");
```

Comment:
@Covertness Do we need to requeue the response here, sort of like like on line 122?

If sending the response fails, how do we retry it?